### PR TITLE
Replace scala.util.Try with a try statement in the DBR buildinfo [databricks]

### LINKED ISSUE #10231: Improve the Maven distro download workaround
Nit: Should the .tar.gz be deleted here?

_Originally posted by @jlowe in https://github.com/NVIDIA/spark-rapids/pull/10225#discussion_r1459141202_

### LINKED ISSUE #11465: [BUG] java.lang.NoClassDefFoundError: org/apache/spark/BuildInfo$ in non-databricks environment
We are seeing a case where an apache spark cluster is misdetecting the Spark version and it is executing code in `DatabricksShimServiceProvider` specifically for spark330db that is not going to work because we don't have those bits in this environment.

Stack trace:

```
ava.lang.NoClassDefFoundError: org/apache/spark/BuildInfo$
06:20:09  	at com.nvidia.spark.rapids.DatabricksShimServiceProvider$.$anonfun$matchesVersion$1(DatabricksShimServiceProvider.scala:30)
06:20:09  	at scala.runtime.java8.JFunction0$mcZ$sp.apply(JFunction0$mcZ$sp.java:23)
06:20:09  	at scala.util.Try$.apply(Try.scala:213)
06:20:09  	at com.nvidia.spark.rapids.DatabricksShimServiceProvider$.matchesVersion(DatabricksShimServiceProvider.scala:29)
06:20:09  	at com.nvidia.spark.rapids.shims.spark330db.SparkShimServiceProvider.matchesVersion(SparkShimServiceProvider.scala:33)
06:20:09  	at com.nvidia.spark.rapids.ShimLoader$.$anonfun$detectShimProvider$11(ShimLoader.scala:270)
06:20:09  	at com.nvidia.spark.rapids.ShimLoader$.$anonfun$detectShimProvider$11$adapted(ShimLoader.scala:269)
06:20:09  	at scala.collection.immutable.Stream.$anonfun$partition$1(Stream.scala:589)
06:20:09  	at scala.collection.immutable.Stream.$anonfun$partition$1$adapted(Stream.scala:589)
06:20:09  	at scala.collection.immutable.Stream.filterImpl(Stream.scala:506)
06:20:09  	at scala.collection.immutable.Stream.filterImpl(Stream.scala:204)
06:20:09  	at scala.collection.TraversableLike.filter(TraversableLike.scala:395)
06:20:09  	at scala.collection.TraversableLike.filter$(TraversableLike.scala:395)
06:20:09  	at scala.collection.AbstractTraversable.filter(Traversable.scala:108)
06:20:09  	at scala.collection.immutable.Stream.partition(Stream.scala:589)
06:20:09  	at com.nvidia.spark.rapids.ShimLoader$.detectShimProvider(ShimLoader.scala:269)
06:20:09  	at com.nvidia.spark.rapids.ShimLoader$.findShimProvider(ShimLoader.scala:299)
06:20:09  	at com.nvidia.spark.rapids.ShimLoader$.initShimProviderIfNeeded(ShimLoader.scala:92)
06:20:09  	at com.nvidia.spark.rapids.ShimLoader$.getShimClassLoader(ShimLoader.scala:184)
06:20:09  	at com.nvidia.spark.rapids.ShimReflectionUtils$.loadClass(ShimReflectionUtils.scala:28)
06:20:09  	at com.nvidia.spark.rapids.ShimReflectionUtils$.newInstanceOf(ShimReflectionUtils.scala:34)
06:20:09  	at com.nvidia.spark.rapids.ShimLoader$.newDriverPlugin(ShimLoader.scala:335)
06:20:09  	at com.nvidia.spark.SQLPlugin.driverPlugin(SQLPlugin.scala:28)
06:20:09  	at org.apache.spark.internal.plugin.DriverPluginContainer.$anonfun$driverPlugins$1(PluginContainer.scala:47)
06:20:09  	at scala.collection.TraversableLike.$anonfun$flatMap$1(TraversableLike.scala:293)
06:20:09  	at scala.collection.mutable.ResizableArray.foreach(ResizableArray.scala:62)
06:20:09  	at scala.collection.mutable.ResizableArray.foreach$(ResizableArray.scala:55)
06:20:09  	at scala.collection.mutable.ArrayBuffer.foreach(ArrayBuffer.scala:49)
06:20:09  	at scala.collection.TraversableLike.flatMap(TraversableLike.scala:293)
06:20:09  	at scala.collection.TraversableLike.flatMap$(TraversableLike.scala:290)
06:20:09  	at scala.collection.AbstractTraversable.flatMap(Traversable.scala:108)
06:20:09  	at org.apache.spark.internal.plugin.DriverPluginContainer.<init>(PluginContainer.scala:46)
06:20:09  	at org.apache.spark.internal.plugin.PluginContainer$.apply(PluginContainer.scala:210)
06:20:09  	at org.apache.spark.internal.plugin.PluginContainer$.apply(PluginContainer.scala:193)
06:20:09  	at org.apache.spark.SparkContext.<init>(SparkContext.scala:560)
06:20:09  	at org.apache.spark.api.java.JavaSparkContext.<init>(JavaSparkContext.scala:58)
06:20:09  	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
06:20:09  	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
06:20:09  	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
06:20:09  	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
06:20:09  	at py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:247)
06:20:09  	at py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:357)
06:20:09  	at py4j.Gateway.invoke(Gateway.java:238)
06:20:09  	at py4j.commands.ConstructorCommand.invokeConstructor(ConstructorCommand.java:80)
06:20:09  	at py4j.commands.ConstructorCommand.execute(ConstructorCommand.java:69)
06:20:09  	at py4j.ClientServerConnection.waitForCommands(ClientServerConnection.java:182)
06:20:09  	at py4j.ClientServerConnection.run(ClientServerConnection.java:106)
06:20:09  	at java.lang.Thread.run(Thread.java:750)
```

### PR DESCRIPTION
**Pull request by gerashegalov**
Fixes #11465  
Fixes #10231

- Add a repro test to the DBR pre-merge, verifyiably failing pre-PR
- Use a `try` statement to catch all exceptions
- Delete the Maven tarball from /tmp
 
Signed-off-by: Gera Shegalov <gera@apache.org>

### CODE REVIEW

**Comment by gerashegalov** [comment]
build

**Review by abellina — APPROVED** [review]

**Comment by abellina** [comment]
build

**Comment by gerashegalov** [comment]
build

File: jenkins/databricks/test.sh [hunk 1]
```diff
@@ -90,6 +91,18 @@ run_pyarrow_tests() {
...
 ## Separate the integration tests into "CI_PART1" and "CI_PART2", run each part in parallel on separate Databricks clusters to speed up the testing process.
 if [[ $TEST_MODE == "DEFAULT" || $TEST_MODE == "CI_PART1" ]]; then
+    # Run two-shim smoke test with the base Spark build
+    if [[ "$WITH_DEFAULT_UPSTREAM_SHIM" != "0" ]]; then
+        if [[ ! -d $HOME/spark-3.2.0-bin-hadoop3.2 ]]; then
+            wget https://archive.apache.org/dist/spark/spark-3.2.0/spark-3.2.0-bin-hadoop3.2.tgz -P /tmp
+            tar xf /tmp/spark-3.2.0-bin-hadoop3.2.tgz -C $HOME
+            rm -f /tmp/spark-3.2.0-bin-hadoop3.2.tgz
+        fi
+        SPARK_HOME=$HOME/spark-3.2.0-bin-hadoop3.2 \
```
Comment (NvTimLiu) [inline jenkins/databricks/test.sh:101]: Is 3.2.0 base spark mandatory? 

This changes SPARK_HOME from DB spark runtime to the 3.2.0 base spark

We should change SPARK_HOME back to DB runtime after the SMOKE test, to make sure the remaining tests are against DB spark runtime.

Comment (gerashegalov) [reply jenkins/databricks/test.sh:101]: > Is 3.2.0 base spark mandatory?

Yes, the test with a multi-shim Plugin including the DBR support but running in a non-DBR Spark runtime is exactly what is missing in the pre-merge pipeline. And we also made it hard #11428


> We should change SPARK_HOME back to DB runtime after the SMOKE test, to make sure the remaining tests are against DB spark runtime.

 Note the `\` for the line continuation. The scope of the environment on L101-103 is just the command on L104.

**Review by sameerz — APPROVED** [review]

**Review by NvTimLiu — APPROVED** [review]

**Pull request merged**